Bug: Wrong file name in error messages during multi-file evaluation

## Description

Error messages sometimes display the wrong file name (typically showing `main.ez` instead of the actual source file), even though the line and column numbers are correct for the actual file where the error occurred.

## Minimal Reproduction

**Directory structure:**
```
test_file_error/
├── main/
│   └── main.ez
└── lib/
    └── helpers.ez
```

**main/main.ez:**
```ez
module main

import @std
import lib"../lib"

do main() {
    std.println("Starting test...")
    lib.trigger_error()
    std.println("Done")
}
```

**lib/helpers.ez:**
```ez
module lib

import @arrays

do trigger_error() {
    temp arr [int] = {1, 2, 3}

    for_each item in arr {
        if item == 2 {
            arrays.remove_at(arr, 0)  // ERROR IS HERE - line 10
        }
    }
}

do another_func() -> int {
    return 42
}
```

**Run:** `ez main/main.ez`

## Expected Behavior

```
error[E9006]: arrays.remove_at() cannot modify array during for_each iteration
  --> lib/helpers.ez:10:13
```

## Actual Behavior

```
error[E9006]: arrays.remove_at() cannot modify array during for_each iteration
  --> main/main.ez:10:19
```

The error points to `main/main.ez:10` which is just `}` - completely wrong file!

## Root Cause Analysis

In `pkg/interpreter/evaluator.go`, the error file location comes from `globalEvalContext.CurrentFile`:

```go
func newErrorWithLocation(code string, line, column int, format string, a ...interface{}) *Error {
    file := ""
    if globalEvalContext != nil {
        file = globalEvalContext.CurrentFile
    }
    // ...
}
```

The `CurrentFile` is updated during multi-file module evaluation but gets out of sync when:
1. Evaluating expressions in imported modules
2. Errors occur in nested function calls across modules
3. The token's `File` field is empty (not set by parser for all contexts)

## Files to Investigate

- `pkg/interpreter/evaluator.go` - `newErrorWithLocation()` and `globalEvalContext.CurrentFile` tracking
- `pkg/parser/parser.go` - Token file assignment (lines 389-391)
- `pkg/interpreter/evaluator.go` - `getStatementFile()` function

## Related

This is NOT specific to E9006 - it can happen with any runtime error in multi-file projects.

--- a/pkg/interpreter/evaluator.go
+++ b/pkg/interpreter/evaluator.go
@@ -1919,6 +1919,10 @@ func evalSingleCast(value Object, targetType string, line, col int) Object {
 			errObj.Line = line
 			errObj.Column = col
 		}
+		// Set file from current context if not already set (#1094)
+		if errObj.File == "" && globalEvalContext != nil {
+			errObj.File = globalEvalContext.CurrentFile
+		}
 	}
 	return result
 }
@@ -2961,6 +2965,10 @@ func evalMemberCall(member *ast.MemberExpression, args []ast.Expression, env *En
 				errObj.Line = member.Token.Line
 				errObj.Column = member.Token.Column
 			}
+			// Set file from current context if not already set (#1094)
+			if errObj.File == "" && globalEvalContext != nil {
+				errObj.File = globalEvalContext.CurrentFile
+			}
 		}
 		return result
 	}
@@ -3077,6 +3085,10 @@ func applyFunction(fn Object, args []Object, line, col int) Object {
 			if errObj.Line == 0 && errObj.Column == 0 {
 				errObj.Line = line
 				errObj.Column = col
+			}
+			// Set file from current context if not already set (#1094)
+			if errObj.File == "" && globalEvalContext != nil {
+				errObj.File = globalEvalContext.CurrentFile
 			}
 		}
 		return result

